server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala

```diff
@@ -38,19 +44,45 @@ class AddressController @Inject()(
     * @param format requested format of the query (paf/nag)
     * @return Json response with addresses information
     */
-  def addressQuery(input: String, format: String): Action[AnyContent] = Action async { implicit req =>
-    logger info s"#addressQuery:\ninput $input , format: $format"
-    input.toOption map { actualInput =>
-      val tokens = parser tag actualInput
-      logger info s"#addressQuery parsed:\n${tokens.map(t => s"value: ${t.value} , label:${t.label}").mkString("\n")}"
-      formatQuery[AddressBySearchResponseContainer, Seq[CrfTokenResult]](
-        formatStr = format,
-        inputForPafFn = AddressQueryInput(tokens),
-        pafFn = pafSearch,
-        inputForNagFn = AddressQueryInput(tokens),
-        nagFn = nagSearch
-      ) getOrElse futureJsonBadRequest(UnsupportedFormat)
-    } getOrElse futureJsonBadRequest(EmptySearch)
+  def addressQuery(input: String, format: String, offset: Option[String] = None, limit: Option[String] = None): Action[AnyContent] = Action async { implicit req =>
+    logger info s"#addressQuery:\ninput $input , format: $format , offset: ${offset.getOrElse("default")}, limit: ${limit.getOrElse("default")}"
+   // get the defaults and maxima for the paging parameters from the config
+    val defLimit = conf.config.elasticSearch.defaultLimit
+    val defOffset = conf.config.elasticSearch.defaultOffset
+    val maxLimit = conf.config.elasticSearch.maximumLimit
+    val maxOffset = conf.config.elasticSearch.maximumOffset
+    val limval = limit.getOrElse(defLimit.toString())
+    val offval = offset.getOrElse(defOffset.toString())
+    val limitInvalid = Try(limval.toInt).isFailure
+    val offsetInvalid = Try(offval.toInt).isFailure
+    val limitInt = Try(limval.toInt).toOption.getOrElse(defLimit)
+    val offsetInt = Try(offval.toInt).toOption.getOrElse(defOffset)
+// Check the offset and limit parameters before proceeding with the request
+    if (limitInvalid){
+      futureJsonBadRequest(LimitNotNumeric)
```

Comment:
I'll give it a try. However, the downside with returning Play's built-in response (probably action not found - will see) rather than the APIs bespoke error response is that it is unfriendly to the client. In my experience API users like a predictable response format, so giving them different response types for different ways in which the parameter is wrong (e.g. limit=0 and limit=ten) might be unpopular.